### ISSUE TITLE
S3 corrupted backup

### ISSUE THREAD

**Issue by mkdel** [2ndquadrant-it/barman#286]
Hi
```
2020-05-22 20:32:55,150 INFO Uploading prodserver/base/20200522T150729/data.tar.gz, part 891 (worker 1)
2020-05-22 20:33:06,811 INFO Uploading prodserver/base/20200522T150729/data.tar.gz, part 892 (worker 0)
2020-05-22 20:33:41,670 INFO Uploading prodserver/base/20200522T150729/data.tar.gz, part 893 (worker 1)
2020-05-22 20:34:16,251 INFO Uploading prodserver/base/20200522T150729/data.tar.gz, part 894 (worker 0)
2020-05-22 20:34:26,794 INFO Uploading prodserver/base/20200522T150729/data.tar.gz, part 895 (worker 1)
2020-05-22 20:34:35,701 INFO S3UploadController.add_file('pg_control', '/var/lib/pgsql/9.6/data/global/pg_control', 'data', 'global/pg_control', False)
2020-05-22 20:34:35,702 INFO Stopping backup 20200522T150729
2020-05-22 20:34:39,768 INFO S3UploadController.add_fileobj('backup_label', 'data', 'backup_label')
2020-05-22 20:34:39,769 INFO S3UploadController.close()
2020-05-22 20:34:39,771 INFO Uploading prodserver/base/20200522T150729/data.tar.gz, part 896 (worker 0)
2020-05-22 20:34:40,144 INFO Completing prodserver/base/20200522T150729/data.tar.gz (worker 1)
2020-05-22 20:34:40,528 INFO S3UploadController.statistics()
2020-05-22 20:34:40,564 INFO Uploading prodserver/base/20200522T150729/backup.info
2020-05-22 20:34:40,639 INFO Backup end at LSN: 86/F30A39C8 (0000000100000086000000F3, 000A39C8)
2020-05-22 20:34:40,640 INFO Backup completed (start time: 2020-05-22 15:07:29.492276, elapsed time: 5 hours, 27 minutes, 11 seconds)
2020-05-22 20:34:40,641 INFO Upload process stopped (worker 0)
2020-05-22 20:34:40,641 INFO Upload process stopped (worker 1)
```

Here is content of backup.info:
```
backup_label='START WAL LOCATION: 86/F3000060 (file 0000000100000086000000F3)\nCHECKPOINT LOCATION: 86/F3000098\nBACKUP METHOD: streamed\nBACKUP FROM: master\nSTART TIME: 2020-05-22 15:07:29 BST\nLABEL: Barman backup None 20200522T150729\n'
begin_offset=96
begin_time=2020-05-22 15:07:29.350613+01:00
begin_wal=0000000100000086000000F3
begin_xlog=86/F3000060
config_file=/var/lib/pgsql/9.6/data/postgresql.conf
copy_stats={'total_time': 19631.358738, 'number_of_workers': 2, 'serialized_copy_time': 502.494439, 'serialized_copy_time_per_item': {'data': 502.494439}, 'analysis_time': 0, 'copy_time': 19602.700657, 'analysis_time_per_item': {'data': 0}, 'copy_time_per_item': {'data': 19602.700657}}
deduplicated_size=None
end_offset=670152
end_time=2020-05-22 20:34:35.703513+01:00
end_wal=0000000100000086000000F3
end_xlog=86/F30A39C8
error=None
hba_file=/var/lib/pgsql/9.6/data/pg_hba.conf
ident_file=/var/lib/pgsql/9.6/data/pg_ident.conf
included_files=None
mode=None
pgdata=/var/lib/pgsql/9.6/data
server_name=None
size=None
status=STARTED
systemid=6641166666224371149
tablespaces=None
timeline=1
version=90617
xlog_segment_size=16777216
```

There suppose to be about 233G in ./base, but actual size of unpacked is : 61G  ./base

**Comment by mnencia** [comment]
Hi, could you please attach the file list returned from `tar ztvf` on the backup tar and the one returned from `find /var/lib/pgsql/9.6/data -ls` on your postgres server?

**Comment by mkdel** [comment]
compared number of files:
```
tar -tvf data.tar.gz | wc -l
gzip: stdin: unexpected end of file
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now
21089
```
VS
`[~/9.6/data]$ find . | wc -l
8857
`
[dataContent.txt](https://github.com/2ndquadrant-it/barman/files/4677820/dataContent.txt)

**Comment by mnencia** [comment]
could you please attach the actual output of `tar -tvf data.tar.gz`?

Thanks,
Marco

**Comment by mkdel** [comment]
Hi Marco,
sure, here
[data.tar.out.txt](https://github.com/2ndquadrant-it/barman/files/4677989/data.tar.out.txt)

btw:
```
 tar -tvf data.tar.gz > /tmp/data.tar.out.txt
gzip: stdin: unexpected end of file
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now
```

**Comment by mnencia** [comment]
I've tried several times without success to reproduce this. If you still have the files I would be interested to see how big is the `data.tar.gz` file. However, I'm inclined to classify thins as an environmental error.

From what is ee the size of `data.tar.gz` file should be around 17.4 GB. (895 parts of 20MB). if the size is a lot less, it should be due to a broken download or some error in AWS. In any case, the fix for #284 should help also in this case.

**Comment by mnencia** [comment]
I believe that after 8a9d8e60cff6224fee0459b0cd43d2b37d432194 this issue could be closed
